osbs/tekton.py

```diff
@@ -720,10 +720,13 @@ def wait_for_taskruns(self):
                     "Pipeline run '%s' does not have any task runs yet",
                     self.pipeline_run_name)
                 continue
+            current_task_runs = []
             for task_run_name, task_run_data in task_runs.items():
                 if task_run_name not in watched_task_runs:
                     watched_task_runs.add(task_run_name)
-                    yield task_run_data['pipelineTaskName'], task_run_name
+                    current_task_runs.append((task_run_data['pipelineTaskName'], task_run_name))
```

Comment:
Is it possible that, when running the 4 parallel tasks, they won't start at the same time and `current_task_runs` will only include some of them?

Reply:
yes, it seems like it is possible

Reply:
not a 100% sure

Reply:
I don't think it is about if those tasks start at the same time, but if they appear in ['status']['taskRuns'] at the same moment, in some new state

which should most likely happen, I'd try this out, since this is at least simple solution

Reply:
I've stopped believing that Tekton will do the reasonable thing, but +1 to test the current solution :smile: